Source/Private/ASTipNode.m

```diff
@@ -0,0 +1,32 @@
+//
+//  ASTipNode.m
+//  AsyncDisplayKit
+//
+//  Created by Adlai Holler on 4/12/17.
+//  Copyright © 2017 Facebook. All rights reserved.
+//
+
+#import "ASTipNode.h"
+
+#if AS_ENABLE_TIPS
+
+// Prevent compiler from complaining about undeclared selector.
+@interface UIResponder (TipNodeActions)
+- (void)didTapTipNode:(id)sender;
+@end
+
+@implementation ASTipNode
+
+- (instancetype)initWithTip:(ASTip *)tip
+{
+  if (self = [super init]) {
+    self.backgroundColor = [UIColor colorWithRed:0 green:0.7 blue:0.2 alpha:0.3];
+    _tip = tip;
+    [self addTarget:nil action:@selector(didTapTipNode:) forControlEvents:ASControlNodeEventTouchUpInside];
```

Comment:
Yep! This is the best-kept secret of UIKit – you can set target nil and use the responder chain, instead of tightly binding all your controls to their controllers.